core/src/main/scala/chisel3/connectable/package.scala

```diff
@@ -15,8 +15,7 @@ package object connectable {
     *
     * @param consumer the left-hand-side of the connection
     */
-  implicit class ConnectableOperators[T <: Data](consumer: T)
-      extends Connectable.ConnectableOpExtension(Connectable(consumer))
+  class ConnectableOperators[T <: Data](consumer: Connectable[T]) extends Connectable.ConnectableOpExtension(consumer)
```

Comment:
Is this working because rather than this class providing the connectable methods to Data, instead the compiler is using the implicit conversion? If so, while I appreciate the frugality of the diff, I worry a bit about the direction it's moving the code. In an ideal world, these methods would be implemented as extension methods and there would be no implicit conversion. Incidentally, the comment on the implicit conversion does not make sense to me:
```
    * Originally this was done with an implicit class, but all functions we want to
    *  add to Data we also want on Connectable, so an implicit conversion makes the most sense
    *  so the ScalaDoc can be shared.
```
This should already be accomplished by the implicit class (the extension methods) extending `ConnectableOpExtension`.

I suspect there is a bigger problem than this though, see comment on the implicit conversion itself.

Reply:
In the latest revision, I've undone this change.